### ISSUE TITLE
`missing_docs` lint does not take `doc(hidden)` into account on re-exports

### ISSUE THREAD

**Issue by Nemo157** [rust-lang/rust#85522]
I tried [this code](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=468c05b2ae3d24063d5a1ae107393f8f):

```rust
#![warn(missing_docs)]

//! Crate docs

mod foo {
    pub struct Foo;
}

#[doc(hidden)]
pub mod bar { pub use crate::foo::Foo; }
```

I expected to see this happen: no warning, as `Foo` is not a publicly documented type

Instead, this happened:

```
warning: missing documentation for a struct
 --> src/lib.rs:6:5
  |
6 |     pub struct Foo;
  |     ^^^^^^^^^^^^^^^
  |
note: the lint level is defined here
 --> src/lib.rs:1:9
  |
1 | #![warn(missing_docs)]
  |         ^^^^^^^^^^^^

warning: 1 warning emitted
```

Testing similar code where there is no re-export correctly identifies that the ancestor module is not publicly visible, only with this setup where the type is re-exported from a `doc(hidden)` module does it fail.

## Meta

This has been the behavior since 1.0.0 (with appropriate edition 2015 changes), up through the current nightly.
